### ISSUE TITLE
Genie Python/IOC: check_limit_violation not reporting soft limit violation

### ISSUE THREAD

**Issue by John-Holt-Tessella** [ISISComputingGroup/IBEX#1783]
1. Set up a motor
1. Put it into a soft limit violation
1. Issue `waitfor_move` - should report limit violation
1. Issue `check_limit_violation` - no violation reported

It should be reporting the violations; fix this.

**Comment by John-Holt-Tessella** [comment]
This seems to work:

g.check_limit_violations("Motor1", "Motor2")
Out[20]: [u'Motor1', u'Motor2']

but it does not report hard limits should it? 

NB if we change the meaning we will need to change docs

**Comment by ISISBuildServer** [comment]
Yes it should return those, but whether we should distinguish the two I don't know.


________________________________

**Comment by John-Holt-Tessella** [comment]
Hard limit should not be returned by default but a flag should be added to allow hard limits to be returned.
